src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -600,92 +641,95 @@ private void Dispose(bool disposing)
-                // We want to find 64-bit SDK because it is more likely to be installed.
-                defaultArchitecture = Environment.Is64BitOperatingSystem ? Architecture.X64 : Architecture.X86;
+            // We are running in vstest.console.exe that was built against .NET
+            // Framework. This console prefers 32-bit because it needs to run as 32-bit
+            // to be compatible with QTAgent. It runs as 32-bit both under VS and in
+            // Developer console. Set the default architecture based on the OS
+            // architecture, to find 64-bit dotnet SDK when running AnyCPU dll on 64-bit
+            // system, and 32-bit SDK when running AnyCPU dll on 32-bit OS.
+            // We want to find 64-bit SDK because it is more likely to be installed.
+            defaultArchitecture = Environment.Is64BitOperatingSystem ? Architecture.X64 : Architecture.X86;
 #endif
-                EqtTrace.Verbose($"TestRequestManager.UpdateRunSettingsIfRequired: Default architecture: {defaultArchitecture} IsDefaultTargetArchitecture: {RunSettingsHelper.Instance.IsDefaultTargetArchitecture}, Current process architecture: {_processHelper.GetCurrentProcessArchitecture()}.");
-            }
+            EqtTrace.Verbose($"TestRequestManager.UpdateRunSettingsIfRequired: Default architecture: {defaultArchitecture} IsDefaultTargetArchitecture: {RunSettingsHelper.Instance.IsDefaultTargetArchitecture}, Current process architecture: {_processHelper.GetCurrentProcessArchitecture()}.");
+        }
 
-            settingsUpdated |= UpdatePlatform(
-                document,
-                navigator,
-                sources,
-                sourcePlatforms,
-                defaultArchitecture,
-                out Architecture chosenPlatform);
+        settingsUpdated |= UpdatePlatform(
+            document,
+            navigator,
+            sources,
+            defaultArchitecture,
+            out Architecture chosenPlatform,
+            out sourceToArchitectureMap);
+
+        if (!FeatureFlag.Instance.IsEnabled(FeatureFlag.MULTI_TFM_RUN))
+        {
             CheckSourcesForCompatibility(
                 chosenFramework,
                 chosenPlatform,
                 defaultArchitecture,
-                sourcePlatforms,
-                sourceFrameworks,
+                sourceToArchitectureMap,
+                sourceToFrameworkMap,
                 registrar);
-            settingsUpdated |= UpdateDesignMode(document, runConfiguration);
-            settingsUpdated |= UpdateCollectSourceInformation(document, runConfiguration);
-            settingsUpdated |= UpdateTargetDevice(navigator, document);
-            settingsUpdated |= AddOrUpdateConsoleLogger(document, runConfiguration, loggerRunSettings);
 
-            updatedRunSettingsXml = navigator.OuterXml;
+            // TODO: NOMERGE: revert all architectures and frameworks on our map to the one common framework and architecture.
```

Comment:
This will allow us to do the same thing as before with FeatureFlag.MULTI_TFM_RUN = 0

Reply:
Did that.